samples/trusted-ai/trusted-ai.py

```diff
@@ -35,56 +29,40 @@ def trusted_ai(
-    objectiveConfig = {
-      "type": "maximize",
-      "goal": goal,
-      "objectiveMetricName": "accuracy",
-      "additionalMetricNames": []
-    }
-    algorithmConfig = {"algorithmName" : "random"}
-    parameters = [
-      {"name": "--dummy", "parameterType": "int", "feasibleSpace": {"min": "1", "max": "2"}},
-    ]
-    rawTemplate = {
-      "apiVersion": "batch/v1",
-      "kind": "Job",
-      "metadata": {
-         "name": "{{.Trial}}",
-         "namespace": "{{.NameSpace}}"
-      },
-      "spec": {
-        "template": {
-          "spec": {
-            "restartPolicy": "Never",
-            "containers": [
-              {"name": "{{.Trial}}",
-               "image": "aipipeline/gender-classification:latest",
-               "command": [
-                   "python", "-u", "gender_classification_training.py", "--data_bucket", "mlpipeline", "--result_bucket", "mlpipeline"
-               ]
-              }
-            ],
-            "env": [{'name': 'S3_ENDPOINT', 'value': 'minio-service.kubeflow:9000'}]
-          }
+    job_name = exp_name.name + "-train"
+    job_manifest = {
+        "apiVersion": "batch/v1",
+        "kind": "Job",
+        "metadata": {
+            "name": "train-job",
+            "namespace": namespace
+        },
+        "spec": {
+            "template": {
+                "metadata": {
+                    "annotations": {
+                        "sidecar.istio.io/inject": "false"
+                    }
+                },
+                "spec": {
+                    "restartPolicy": "Never",
+                    "containers": [
+                        {"name": "classification-training",
+                         "image": "aipipeline/gender-classification:latest",
+                         "command": [
+                             "python", "-u", "gender_classification_training.py", "--data_bucket", "mlpipeline",
+                             "--result_bucket", "mlpipeline"
+                         ],
+                         "env": [{'name': 'S3_ENDPOINT', 'value': 'minio-service.kubeflow:9000'}]
+                         }
+                    ],
+                }
+            }
         }
-      }
-    }
-    trialTemplate = {
-      "goTemplate": {
-        "rawTemplate": json.dumps(rawTemplate)
-      }
     }
-    katib_run = katib_experiment_launcher_op(
-        experiment_name=name,
-        experiment_namespace=namespace,
-        parallel_trial_count=parallelTrialCount,
-        max_trial_count=maxTrialCount,
-        objective=str(objectiveConfig),
-        algorithm=str(algorithmConfig),
-        trial_template=str(trialTemplate),
-        parameters=str(parameters),
-        experiment_timeout_minutes=experimentTimeoutMinutes,
-        delete_finished_experiment=deleteAfterDone)
+    train_step = dsl.ResourceOp(name=job_name,
+                                k8s_resource=job_manifest,
+                                action='create')
```

Comment:
I think there's some race condition on this task because you wait the job to be completed before moving to the next task. 

You can set the success condition in resourceOps like this 
https://github.com/kubeflow/kfp-tekton/blob/master/samples/e2e-mnist/e2e-mnist.py#L278

In your case this is a Kubernetes job, so check the Kubernetes documentation on what status is in the job api.